org.eclipse.xtext/src/org/eclipse/xtext/nodemodel/impl/RootNode.java

```diff
@@ -153,7 +153,7 @@ protected void basicSetParent(CompositeNode parent) {
 	}
 	
 	@Override
-	void readData(DataInputStream in, DeserializationConversionContext context) throws IOException {
+	public void readData(DataInputStream in, DeserializationConversionContext context) throws IOException {
```

Comment:
I'd prefer to make this protected rather than public.

Other protected implementation details are made accessible via the InternalNodeModelUtils. You probably want to make readData accessible there such that clients can extend InternalNodeModelUtils to call readData directly.

I'm asking for this since it's strictly an implementation detail how the AbstractNode hierarchy is designed. The indirection via InternalNodeModelUtils gives us at least a little wiggle room for optimizations / refactorings - which didn't happen in the recent past, but there exist ideas on how to reduce the memory footprint of the node model.

The more methods are directly available on that type hierarchy, the harder it gets to make changes.